psycopg/psycopg/server_cursor.py

```diff
@@ -108,6 +108,7 @@ def _close_gen(self, cur: BaseCursor[ConnectionType, Row]) -> PQGen[None]:
                 "SELECT 1 FROM pg_catalog.pg_cursors WHERE name = {}"
             ).format(sql.Literal(self.name))
             res = yield from cur._conn._exec_command(query)
+            assert res is not None, "pipeline mode not supported"  # TODO
```

Comment:
server-side cursors in pipeline mode. mmm... It might be useful but it might pretty complicated maybe?

I'd be tempted to raise `NotSupportedError` to leave the door open to think about it in the future but leave it out of the first implementation.